internal/cli/arguments.go

```diff
@@ -31,6 +34,42 @@ func (a *Argument) Ask(cmd *cobra.Command, value interface{}) error {
 	return askArgument(cmd, a, value)
 }
 
+type pickerOptionsFunc func() (pickerOptions, error)
+
+func (a *Argument) Pick(cmd *cobra.Command, result *string, fn pickerOptionsFunc) error {
+	var opts pickerOptions
+	err := ansi.Waiting(func() error {
+		var err error
+		opts, err = fn()
+		return err
+	})
+
+	if err != nil {
+		return err
+	}
+
+	defaultLabel := opts.defaultLabel()
+	var val string
+	if err := selectArgument(cmd, a, &val, opts.labels(), &defaultLabel); err != nil {
+		return err
+	}
+
+	*result = opts.getValue(val)
+	return nil
+}
+
+func selectArgument(cmd *cobra.Command, a *Argument, value interface{}, options []string, defaultValue *string) error {
+	if canPrompt(cmd) {
+		input := prompt.SelectInput("", a.GetLabel(), a.GetHelp(), options, auth0.StringValue(defaultValue), true)
+
+		if err := prompt.AskOne(input, value); err != nil {
+			return handleInputError(err)
+		}
+	}
+
```

Comment:
Here, in the case of non-interactivity, we need to return an error, like in `askArgument`.